### DISCUSSION TITLE
Creating CRUD app with linked/related tables

### DISCUSSION THREAD

**Discussion by CartCaved** [Budibase/budibase#7474] [Help]
Hello!

Maybe I'm not searching for the right terms but I'm nut understanding how to create an app with linked SQL tables. I'd be very grateful if anyone can give me some general pointer or link to where in the documentation.

This is basically what I'd like to accomplish. Let's say I have three tables in an external existing SQL database - orders, order_rows, suppliers. Each line in the order_rows table had a relationship to a unique order id (one to many relationship). Each order has a supplier from the suppliers table (another one to many).

I essentially want these view:
* a view where I can create a new order, fill out the fields related to that item and also add one or more order lines. I'd like to be able to select a supplier from the linked table or add another supplier.
* When I select an order I want to see the fields, the order rows (table block?) and the supplier name.
* I'd also like an overview of orders where I can see the order with the sum of the related order row quantity * price, for example.

How do I start to create my orders app?

Thanks!

**Comment by insomniacj93** [comment]
@CartCaved
Hi, well guessing that you already have that database done and budibase installed i would suggest these steps.

1- Enter budibase with your account or create one if you didn't yet,
2- After entering click on "Apps" tab on top of your Admin interface,
3- Create the app (give it a name of your choice),
4- Once the app is done, enter in the app "Edit" button to start developing it,
5- Inside the Editing panel, press the "Data" tab,
6- Click the "+" button to add a new Data source,
7- Choose the type of DB you're using,
8- Enter all the information regarding it's location (e.g. if using docker the host should be "host.docker.internal"),
// in my case it was user:"root" password:"root" (but it differs on what you choose in your SQL) and database: "db_test".
9- Press ADD,
10- If all worked as planned you should now have the connection done, next we go to Design tab,
11- On Design tab look for "Add Screen" and when it opens choose the C.R.U.D. option (the second one),
12- Choose from your db the table you want to generate the forms, on creation budibase will generate a 3 screens for each table,
// 1: for insertion/creation of a new row 2: accessing the table and view the rows created 3: access those new rows and allowing you to alter data (also accessable thru the second screen by clicking the "View" button on each row)

Hope i could help you out, if not there is a proper guide in docs.

https://docs.budibase.com/docs/build-a-crud-app-sql

Wish you luck !

**Reply by CartCaved** [reply]
@insomniacj93 
Thank you for the post - very comprehensive. I don't think it really was what I was looking for though, or maybe what I'm looking for is not possible. I'm looking for the steps after 12! :)

I need to create one view where the user can create a new row in one table and 0-many rows in a second table, with the rows in the second table linked to the item being created in the first table.

Think registering an order in a shop. You have one table with all the orders, including the customer, address, order date, payment status, whatever you need to track about the order as a whole.

Then you have a second table containing all the items ordered across all orders.
Orders table
|Order ID|Customer name|Order date|
|---------|----------------|------------|
|Ord1|Funky Frederic|June 17, 2035|
|Ord2|Germane Geraldine|July 1, 2035|

Order items table
|OrderItemID|Order ID| Item name|Quantity|
|-------------|---------|-----------|---|
|OrdItem1|Ord1|Fluffy Floral arrangement|5|
|OrdItem2|Ord1|Citrus Comic|2|
|OrdItem3|Ord2|Fluffy Floral arrangement|2|

I want a view where the user can add a new order and any number of order item lines for that order. And a view that can show the user Order 1 with the two items ordered.

Like this:
Order
|**Order 1**||
|--|--|
|Ordered by| Funky Frederic|
|Date| June 17, 2035|
|_Items_||
|Fluffy Floral arrangements|5|
|Citrus Comic|2|


Is that possible in BudiBase?

**Reply by melohagan** [reply]
Hey @CartCaved 

I'm not entirely certain what you're looking for, but these may help:
- https://docs.budibase.com/docs/filter-by-relationships
- https://docs.budibase.com/docs/fetching-data#displaying-relational-data
